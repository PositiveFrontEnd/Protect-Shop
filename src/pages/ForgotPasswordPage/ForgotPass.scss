.form__container__box__forgot__pass {
  .form__container__box__main__authorization {
    .forgot__pass__title {
      margin-top: 8px;
      margin-bottom: 8px;
      color: var(--black);
      font-size: 18px;
      text-transform: uppercase;
    }

    .forgot__pass__text {
      max-width: 273px;
      margin-top: 16px;
      margin-bottom: 24px;
      color: var(--black);
      font-size: 14px;
      text-align: center;
      text-transform: uppercase;
    }

    .button__forgot__pass__page {
      margin-top: 24px;
      margin-bottom: 32px;
    }
  }

  @media (width >= 768px) {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .form__container__box__main__authorization {
      margin-left: 0;
      .forgot__pass__title {
        margin-top: 20px;
        margin-bottom: 20px;
        color: var(--black);
        font-size: 24px;
      }

      .forgot__pass__text {
        max-width: 384px;
        margin-top: 20px;
        margin-bottom: 28px;
        text-align: center;
      }
    }

    .img__form__forgot__pass__container {
      width: 100%;
    }
  }

  @media (width>=1024px) {
  }
}
