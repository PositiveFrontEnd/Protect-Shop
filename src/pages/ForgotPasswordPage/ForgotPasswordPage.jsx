import React from "react";
import ForgotPasswordForm from "../../components/Form/ForgotPassword/ForgotPasswordF";
import ForgotPassImg from "/Images/forgotPassF.jpg";
import "../AuthorizationPage/AuthorizationPage.scss";
import "../RegistrationPage/RegastrationPage.scss";
import "./ForgotPass.scss";

const ForgotPasswordPage = () => {
  return (
    <>
      <section className="container">
        <div className="form__container__box__forgot__pass form__container__box__authorization ">
          <div className="form__container__box__main__authorization">
            <h2 className="forgot__pass__title">Forgot your password?</h2>
            <p className="forgot__pass__text">
              Enter the E-Mail for your account. Click continue to receive your
              password via email.
            </p>
            <ForgotPasswordForm />
          </div>
          <div className="img__form__authorization__container img__form__forgot__pass__container">
            <img
              className="img__form__authorization img__form__forgot__pass"
              src={ForgotPassImg}
              alt="forgot_pass_img"
            />
          </div>
        </div>
      </section>
    </>
  );
};

export default ForgotPasswordPage;