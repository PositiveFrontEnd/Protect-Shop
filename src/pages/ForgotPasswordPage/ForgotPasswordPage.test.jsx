import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import ForgotPasswordPage from "./ForgotPasswordPage";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  ForgotPasswordPage", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };

        const store = mockStore(initialState);


        const forgotPasswordPage = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ForgotPasswordPage />
                </BrowserRouter>
            </Provider>
        );

        expect(forgotPasswordPage).toMatchSnapshot();
    });
});
