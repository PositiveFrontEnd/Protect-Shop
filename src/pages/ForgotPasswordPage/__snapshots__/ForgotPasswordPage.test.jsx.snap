// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  ForgotPasswordPage > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container"
      >
        <div
          class="form__container__box__forgot__pass form__container__box__authorization "
        >
          <div
            class="form__container__box__main__authorization"
          >
            <h2
              class="forgot__pass__title"
            >
              Forgot your password?
            </h2>
            <p
              class="forgot__pass__text"
            >
              Enter the E-Mail for your account. Click continue to receive your password via email.
            </p>
            <div
              class="form__box-registr"
            >
              <form
                action="#"
              >
                <label
                  class="form__field form__field-modal form__field-registr  mb-3"
                >
                  <div
                    class="form__little__box "
                  >
                    <p
                      class="form__title"
                    >
                      Email
                    </p>
                    <input
                      class="form__input form__input-modal"
                      name="email"
                      placeholder="Email"
                      type="text"
                      value=""
                    />
                  </div>
                </label>
                <button
                  class="button-black button__submit button__forgot__pass__page button__registr__page"
                  type="submit"
                >
                  Continue
                </button>
              </form>
            </div>
          </div>
          <div
            class="img__form__authorization__container img__form__forgot__pass__container"
          >
            <img
              alt="forgot_pass_img"
              class="img__form__authorization img__form__forgot__pass"
              src="/Images/forgotPassF.jpg"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container"
    >
      <div
        class="form__container__box__forgot__pass form__container__box__authorization "
      >
        <div
          class="form__container__box__main__authorization"
        >
          <h2
            class="forgot__pass__title"
          >
            Forgot your password?
          </h2>
          <p
            class="forgot__pass__text"
          >
            Enter the E-Mail for your account. Click continue to receive your password via email.
          </p>
          <div
            class="form__box-registr"
          >
            <form
              action="#"
            >
              <label
                class="form__field form__field-modal form__field-registr  mb-3"
              >
                <div
                  class="form__little__box "
                >
                  <p
                    class="form__title"
                  >
                    Email
                  </p>
                  <input
                    class="form__input form__input-modal"
                    name="email"
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
              </label>
              <button
                class="button-black button__submit button__forgot__pass__page button__registr__page"
                type="submit"
              >
                Continue
              </button>
            </form>
          </div>
        </div>
        <div
          class="img__form__authorization__container img__form__forgot__pass__container"
        >
          <img
            alt="forgot_pass_img"
            class="img__form__authorization img__form__forgot__pass"
            src="/Images/forgotPassF.jpg"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
